File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min_length = Integer.MAX_VALUE;
        int curr_sum = nums[0];
        int start = 0;
        int end = 0;
        
        while (end < nums.length) {
            
            if (curr_sum >= target) {
                if (end - start + 1 < min_length) min_length = end - start + 1;
                curr_sum = curr_sum - nums[start];
                start++;
            }
            
            else {
                end++;
                if (end < nums.length) curr_sum += nums[end];
            }
        }
        
        return min_length == Integer.MAX_VALUE ? 0 : min_length;
    }
}

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        r = 0
        l = 0
        curr_sum = 0
        res = float('inf')
        
        for r in range(len(nums)):
            curr_sum += nums[r]
            while curr_sum >= target:
                res = min(res, r-l+1)
                curr_sum -= nums[l]
                l+=1
                
        return res if res != float('inf') else 0